File: poo/Cpp/HalfAdder/HalfAdder.cpp
#include <iostream>

int main() {
    bool p, q;
    std::cin >> std::boolalpha >> p >> q;
    bool vai_um = p && q;
    bool soma = p != q;
    std::cout << std::boolalpha << vai_um << " " << soma << std::endl;
    return 0;
}

File: poo/Java/HalfAdder/HalfAdder.java
import java.util.Scanner;

public class HalfAdder {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean p = scanner.nextBoolean();
        boolean q = scanner.nextBoolean();
        boolean vaiUm = p && q;
        boolean soma = p ^ q;
        System.out.println(vaiUm + " " + soma);
        scanner.close();
    }
}